src/flows/pipes/Schedule/ExportTaskButton.tsx

```diff
@@ -100,24 +104,41 @@ const ExportTaskButton: FC<Props> = ({
         }
 
         if (isFlagEnabled('createWithFlows')) {
-          fetch(`/api/v2private/notebooks/${flow.id}/resources`, {
-            method: 'POST',
-            headers: {
-              'Content-Type': 'application/json',
-            },
-            body: JSON.stringify({
-              panel: data.id,
-              resource: resp.data.id,
-              type: 'tasks',
-            }),
-          })
-        }
+          new Promise(resolve => {
+            if (flow.id) {
+              resolve(flow.id)
+              return
+            }
 
-        dispatch(notify(taskCreatedSuccess()))
-        dispatch(checkTaskLimits())
+            add(flow).then(id => resolve(id))
+          })
+            .then(id => {
+              return fetch(`/api/v2private/notebooks/${id}/resources`, {
+                method: 'POST',
+                headers: {
+                  'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                  panel: data.id,
+                  resource: resp.data.id,
+                  type: 'tasks',
+                }),
+              }).then(() => {
+                return id
+              })
+            })
+            .then(id => {
+              dispatch(notify(taskCreatedSuccess()))
+              dispatch(checkTaskLimits())
+
+              if (onCreate) {
+                onCreate(resp.data)
+              }
 
-        if (onCreate) {
-          onCreate(resp.data)
+              if (id !== flow.id) {
+                history.replace(`/orgs/${org.id}/notebooks/${id}`)
+              }
+            })
```

Comment:
sorry, my brain is having a hard time. Shouldn't we be updating the flow.list here?

Reply:
the `add(flow)` is updating the list once theres a successful post. this just redirects to the new url so that it's not `/notebooks/from/default` up there when people start clicking around